jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/toolwindow/SqsWindowUI.kt

```diff
@@ -3,20 +3,38 @@
 
 package software.aws.toolkits.jetbrains.services.sqs.toolwindow
 
-import com.intellij.openapi.project.Project
 import com.intellij.ui.components.JBTabbedPane
+import com.intellij.util.ui.JBUI
+import software.amazon.awssdk.services.sqs.SqsClient
 import software.aws.toolkits.jetbrains.services.sqs.Queue
 import software.aws.toolkits.resources.message
+import javax.swing.event.ChangeListener
 
 // Will add more parameters once window is populated
-class SqsWindowUI(private val project: Project, val queue: Queue) {
+class SqsWindowUI(
+    private val client: SqsClient,
+    val queue: Queue
+) {
+    private val pane = PollMessagePane(client, queue)
+    // This makes sure that PollMessagePane is properly set up when switched to the Poll Message tab
+    private val changeListener = ChangeListener { e ->
+        val selected = e?.source as JBTabbedPane
+        if ((selected.selectedIndex == POLL_MESSAGE_PANE) && (!pane.isSetup)) {
```

Comment:
What is the isSetup for? can this be a function called by the thing that creates the table after it is created/in init?